AutoCollection/Statsbeat.ts

```diff
@@ -286,42 +306,45 @@ class Statsbeat {
                 });
                 currentCounter.totalSuccesfulRequestCount = 0; //Reset
             }
-            if (currentCounter.totalFailedRequestCount > 0) {
-                properties = Object.assign({ ...properties, statusCode: this._statusCode });
-                this._statbeatMetrics.push({
-                    name: Constants.StatsbeatCounter.REQUEST_FAILURE,
-                    value: currentCounter.totalFailedRequestCount,
-                    properties: properties
+            if (currentCounter.totalFailedRequestCount.length > 0) {
+                currentCounter.totalFailedRequestCount.forEach((currentCounter) => {
+                    this._statbeatMetrics.push({
+                        name: Constants.StatsbeatCounter.REQUEST_FAILURE,
+                        value: currentCounter.count,
+                        properties: properties
```

Comment:
Where are you using currentCounter.statusCode?, I can see you have an accurate count but I'm not sure where are you actually adding it to the Statsbeat properties

Reply:
Should be fixed now. Added it to the properties in each counter where it's relevant per the spec.